services/publisher/publisher.go

```diff
@@ -499,19 +528,26 @@ func (bd *BatchData) AppendAlert(notif *protocol.NotifyRequest) {
 		}
 	} else if isBlockAlert {
 		blockNum := hexutil.MustDecodeUint64(notif.EvalBlockRequest.Event.BlockNumber)
-		bd.AddBatchAgent(notif.AgentInfo, blockNum, "")
+		bd.AddBatchAgent(notif.AgentInfo, blockNum, "", "")
 		blockRes := bd.GetBlockResults(notif.EvalBlockRequest.Event.BlockHash, blockNum, notif.EvalBlockRequest.Event.Block.Timestamp)
 		if hasAlert {
 			agentAlerts = (*BlockResults)(blockRes).GetAgentAlerts(notif.AgentInfo)
 		}
-	} else {
+	} else if isTxAlert {
 		blockNum := hexutil.MustDecodeUint64(notif.EvalTxRequest.Event.Block.BlockNumber)
-		bd.AddBatchAgent(notif.AgentInfo, blockNum, notif.EvalTxRequest.Event.Receipt.TransactionHash)
+		bd.AddBatchAgent(notif.AgentInfo, blockNum, notif.EvalTxRequest.Event.Receipt.TransactionHash, "")
 		blockRes := bd.GetBlockResults(notif.EvalTxRequest.Event.Block.BlockHash, blockNum, notif.EvalTxRequest.Event.Block.BlockTimestamp)
 		if hasAlert {
 			txRes := (*BlockResults)(blockRes).GetTransactionResults(notif.EvalTxRequest.Event)
 			agentAlerts = (*TransactionResults)(txRes).GetAgentAlerts(notif.AgentInfo)
 		}
+	} else if isCombinationAlert {
+		// TODO REMOVE THIS BEFORE PRODUCTION
```

Comment:
Do we need to remove something here? I think we would append the bots here somehow when they scan "anything".